### PR TITLE
Fix: add custom network with metamask

### PR DESCRIPTION
**Pull request by Captain-Tab**

### CODE REVIEW

File: packages/eth-network/src/CustomNetwork/NewCustomNetworkModal.js [hunk 1]
```diff
@@ -46,6 +46,31 @@ export default class CustomNetworkModal extends PureComponent {
     this.setState({ pending: false })
   }
 
+  getNewNetList = () => {
+    const customNetworkMap = redux.getState().customNetworks.toJS()
+    const customNetworkGroup = Object.keys(customNetworkMap).map(name => ({
+      group: 'others',
+      icon: 'fas fa-vial',
+      id: name,
+      networkId: name,
```
Comment (dandansad) [inline packages/eth-network/src/CustomNetwork/NewCustomNetworkModal.js:55]: need Change to : 
networkId: customeNetworkMap[name]?.networkId || name

File: packages/eth-network/src/networkManager.js [hunk 2]
```diff
@@ -97,68 +102,68 @@ class NetworkManager {
...
   }
 
-  async setNetwork(network, { force, redirect = true, notify = true } = {}) {
+  async requestMetaMaskRPC(networkInfo) {
+    const metaMaskClient = this.browserExtension.ethereum
+    const currentChainId = this.browserExtension.getChainId
+      ? await this.browserExtension.getChainId()
+      : await metaMaskClient.request({ method: 'eth_chainId' })
+    if (!networkInfo.chainId || currentChainId === networkInfo.chainId ) return
+    const hexChainId = `0x${(+networkInfo.chainId).toString(16)}`
```
Comment (dandansad) [inline packages/eth-network/src/networkManager.js:115]: + what does this mean?
(+networkInfo.chainId)

File: packages/eth-i18n/src/locales/zh.js [hunk 3]
```diff
@@ -115,6 +115,9 @@ const translation = {
       delConTips: '它将被销毁。',
       unsupportedNetwork: '不支持此网络',
       unsupportedNetworkText: '请在区块链浏览器上查看.',
+      duplicatedTitle: '添加网络失败',
+      duplicatedText: '参数RPC url<b>{{chainId}}</b>已存在。',
+      lackChainId: 'chainId参数无效, 请检查chainId参数'
```
Comment (iNorthIsle) [inline packages/eth-i18n/src/locales/zh.js:120]:
```suggestion
      duplicatedText: '参数 RPC URL <b>{{chainId}}</b>已存在。',
      lackChainId: 'chainId 参数无效, 请检查 chainId 参数'
```

File: packages/eth-network/src/networkManager.js [hunk 4]
```diff
@@ -97,68 +102,71 @@ class NetworkManager {
...
+    const metaMaskClient = this.browserExtension.ethereum
+    const currentChainId = this.browserExtension.getChainId
+      ? await this.browserExtension.getChainId()
+      : await metaMaskClient.request({ method: 'eth_chainId' })
+    if (currentChainId === networkInfo.chainId) return
+    if (!networkInfo.chainId) {
+      notification.error(t('network.custom.err'), t('network.custom.lackChainId'))
+      return 
+    }
+    const hexChainId = `0x${(+networkInfo.chainId).toString(16)}`
```
Comment (iNorthIsle) [inline packages/eth-network/src/networkManager.js:119]: can use the util method of ethers
https://docs.ethers.io/v5/api/utils/bytes/#utils-hexlify

**Review by iNorthIsle — APPROVED** [review]

**Pull request merged**